### PR TITLE
MOB-4565 checkout clean up old implementation

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
## JIRA
[MOB-4565](https://karhoo.atlassian.net/browse/MOB-4565)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4565]: https://karhoo.atlassian.net/browse/MOB-4565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: KarhooUISDK/Builders/CheckoutScreenBuilder.swift [hunk 1]
```diff
@@ -11,22 +11,24 @@ import KarhooSDK
...
         callback: @escaping ScreenResultCallback<KarhooCheckoutResult>
-    ) -> Screen
+    ) -> KarhooUISDKSceneCoordinator
 }
 
 public extension CheckoutScreenBuilder {
     func buildCheckoutScreen(
-        quote: Quote,
+        navigationController: UINavigationController?,
+        quote: KarhooSDK.Quote,
```
Comment (dianakarh) [inline KarhooUISDK/Builders/CheckoutScreenBuilder.swift:25]: Why not just `import KarhooSDK` instead of adding it everywhere we use something from it?

Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/Builders/CheckoutScreenBuilder.swift]: Great catch. It was an old, temporary, implementation.

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -256,11 +252,11 @@
...
 		51219495290BDC230090E908 /* ExampleSnapshotSpec.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51219494290BDC230090E908 /* ExampleSnapshotSpec.swift */; };
-		51255B102966EAA700290A74 /* NewCheckoutViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B0F2966EAA700290A74 /* NewCheckoutViewController.swift */; };
-		51255B122966EB8700290A74 /* NewCheckout+MVPC.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B112966EB8700290A74 /* NewCheckout+MVPC.swift */; };
-		51255B14296731A400290A74 /* NewCheckoutViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B13296731A400290A74 /* NewCheckoutViewModel.swift */; };
-		51255B1629673AC100290A74 /* NewCheckoutView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B1529673AC100290A74 /* NewCheckoutView.swift */; };
-		51255B1829673D0A00290A74 /* NewCheckoutCoordinator.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B1729673D0A00290A74 /* NewCheckoutCoordinator.swift */; };
+		51255B102966EAA700290A74 /* CheckoutViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51255B0F2966EAA700290A74 /* CheckoutViewController.swift */; };
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:255]: These should have Karhoo as a prefix for consistency.

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -471,6 +465,7 @@
 		51F614D92861B5BB00A7EC11 /* NumericFilterView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51F614D82861B5BB00A7EC11 /* NumericFilterView.swift */; };
 		51F614DB2861B5DE00A7EC11 /* FilterView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51F614DA2861B5DE00A7EC11 /* FilterView.swift */; };
 		51FB611929752F01003FE5AB /* KarhooLegalNoticeViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51FB611829752F01003FE5AB /* KarhooLegalNoticeViewModel.swift */; };
+		51FC1216298D533F00DE9019 /* QuoteVehicleCapacityView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 51FC1215298D533F00DE9019 /* QuoteVehicleCapacityView.swift */; };
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:468]: KarhooQuoteVehicleCapacityView

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -580,18 +555,16 @@
...
 		7BA696DB291BF9CF00A4A1FE /* DetailsCellView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BA696D9291BF9CF00A4A1FE /* DetailsCellView.swift */; };
 		7BA990A729780E31003E9010 /* KarhooBottomSheetCommentsView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BA990A629780E31003E9010 /* KarhooBottomSheetCommentsView.swift */; };
 		7BA990A929780E7D003E9010 /* KarhooBottomSheetCommentsViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BA990A829780E7D003E9010 /* KarhooBottomSheetCommentsViewModel.swift */; };
-		7BC87EAD298988C6008C5120 /* NewLoyaltyViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BC87EAC298988C6008C5120 /* NewLoyaltyViewModel.swift */; };
-		7BE8B2BD2982B68D005E8495 /* LoyaltyEarnBurnView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BE8B2BC2982B68D005E8495 /* LoyaltyEarnBurnView.swift */; };
+		7BC87EAD298988C6008C5120 /* LoyaltyViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BC87EAC298988C6008C5120 /* LoyaltyViewModel.swift */; };
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:564]: KarhooLoyaltyViewModel

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -580,18 +555,16 @@
...
+		7BC87EAF29898931008C5120 /* (null) in Sources */ = {isa = PBXBuildFile; };
+		7BE8B2BD2982B68D005E8495 /* LoyaltyView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7BE8B2BC2982B68D005E8495 /* LoyaltyView.swift */; };
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:566]: KarhooLoyaltyView

File: KarhooUISDK/Builders/CheckoutScreenBuilder.swift [hunk 6]
```diff
@@ -11,22 +11,24 @@ import KarhooSDK
...
     func buildCheckoutScreen(
+        navigationController: UINavigationController?,
         quote: Quote,
         journeyDetails: JourneyDetails,
         bookingMetadata: [String: Any]? = nil,
         callback: @escaping ScreenResultCallback<KarhooCheckoutResult>
-    ) -> Screen {
-        
-        return buildCheckoutScreen(
+    ) -> KarhooUISDKSceneCoordinator {
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Builders/CheckoutScreenBuilder.swift:29]: previously this method returned `Screen` as a result, so `buildCheckoutScreen` was a good name. Now it returns Coordinator, maybe you can think about better naming for this method?

**Review by dianakarh — APPROVED** [review]

**Pull request merged**